src/refnux.coffee

```diff
@@ -50,6 +50,15 @@ createStore = (state = {}) ->
         # execute the action
         try
             newval = action state, dispatch
+
+            if newval and typeof(newval.then) is 'function'
+                dispatching = false
+                return newval.then(
```

Comment:
it's one or the other. what do you think is best?

but then the promise version needs to do the same...

maybe the entire state. somewhat more useful that a partial state?